app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt

```diff
@@ -90,322 +90,403 @@ class TopicPracticeFragmentTest {
+      onView(
+        atPositionOnView(
+          R.id.topic_practice_skill_list,
+          5,
+          R.id.topic_practice_start_button
+        )
+      ).check(matches(not(isClickable())))
     }
   }
 
   @Test
-  // TODO(#973): Fix TopicPracticeFragmentTest
-  @Ignore
   fun testTopicPracticeFragment_loadFragment_selectSubtopics_isSuccessful() {
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
+      testCoroutineDispatchers.runCurrent()
       onView(
         allOf(
           withText(TopicTab.getTabForPosition(2).name),
           isDescendantOfA(withId(R.id.topic_tabs_container))
         )
       ).perform(click())
+      testCoroutineDispatchers.runCurrent()
       onView(
         atPositionOnView(
           R.id.topic_practice_skill_list,
-          1, R.id.subtopic_check_box
+          1,
+          R.id.subtopic_check_box
         )
-      ).perform(
-        click()
-      )
+      ).perform(click())
+      onView(
+        atPositionOnView(
+          R.id.topic_practice_skill_list,
+          1,
+          R.id.subtopic_check_box
+        )
+      ).check(matches(isChecked()))
       onView(
         atPositionOnView(
           R.id.topic_practice_skill_list,
           2, R.id.subtopic_check_box
         )
-      ).perform(
-        click()
-      )
+      ).perform(click())
+      onView(
+        atPositionOnView(
+          R.id.topic_practice_skill_list,
+          2, R.id.subtopic_check_box
+        )
+      ).check(matches(isChecked()))
     }
   }
 
   @Test
-  // TODO(#973): Fix TopicPracticeFragmentTest
-  @Ignore
   fun testTopicPracticeFragment_loadFragment_selectSubtopics_startButtonIsActive() {
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
+      testCoroutineDispatchers.runCurrent()
       onView(
         allOf(
           withText(TopicTab.getTabForPosition(2).name),
           isDescendantOfA(withId(R.id.topic_tabs_container))
         )
       ).perform(click())
+      testCoroutineDispatchers.runCurrent()
       onView(
         atPositionOnView(
           R.id.topic_practice_skill_list,
-          1, R.id.subtopic_check_box
+          1,
+          R.id.subtopic_check_box
         )
-      ).perform(
-        click()
-      )
+      ).perform(click())
+      testCoroutineDispatchers.runCurrent()
       onView(withId(R.id.topic_practice_skill_list)).perform(
         scrollToPosition<RecyclerView.ViewHolder>(
-          10
+          5
         )
       )
-      onView(withId(R.id.topic_practice_start_button)).check(matches(isClickable()))
+      onView(
+        atPositionOnView(
+          R.id.topic_practice_skill_list,
+          5,
+          R.id.topic_practice_start_button
+        )
+      ).check(matches(isClickable()))
     }
   }
 
   @Test
-  // TODO(#973): Fix TopicPracticeFragmentTest
-  @Ignore
   fun testTopicPracticeFragment_loadFragment_selectSubtopics_deselectSubtopics_isSuccessful() {
```

Comment:
```suggestion
  fun testTopicPracticeFragment_loadFragment_selectSubtopics_thenDeselect_selectsCorrectTopic() {
```

This is maybe slightly clearer & better matches your changes.

Reply:
done